src/cohere/manually_maintained/tokenizers.py

```diff
@@ -0,0 +1,50 @@
+import requests
+import typing
+from tokenizers import Tokenizer  # type: ignore
+
+if typing.TYPE_CHECKING:
+    from cohere.client import AsyncClient, Client
+
+TOKENIZER_CACHE_KEY = "tokenizers"
+
+
+def tokenizer_cache_key(model: str) -> str:
+    return f"{TOKENIZER_CACHE_KEY}:{model}"
+
+
+async def get_hf_tokenizer(co: typing.Union["AsyncClient", "Client"], model: str) -> Tokenizer:
+    """Returns a HF tokenizer from a given tokenizer config URL."""
+    from cohere.client import AsyncClient
```

Comment:
why is this dynamically imported?

Reply:
to fix circular imports

Reply:
it's only used to check if it's an instance of Async/sync to avoid having two different methods